test/PizzeriaTest.cpp

```diff
@@ -65,3 +64,36 @@ TEST_F(PizzeriaTest, calculatePriceForPizzaMock)
 
     delete mock;
 }
+
+TEST(PizzeriaTestMain, shouldCalculateOrderLikeInMainFunction){
+    // Given
+    constexpr double mockFirstPrice = 20.0;
+    constexpr double mockSecondPrice = 25.0;
+    constexpr char mockFirstName[] = "Pizza mock first";
+    constexpr char mockSecondName[] = "Pizza mock second";
+    constexpr minutes mockFirstBakingTime{10};
+    constexpr minutes mockSecondBakingTime{5};
+
+    StrictMock<TimeMock> tm;
+    Pizzeria bravo("Bravo Pizza", tm);
+    StrictMock<PizzaMock> pm1;
+    NiceMock<PizzaMock> pm2;
+    Pizzas pizzas = {new PizzaStub{"stub"}, &pm1, &pm2};
+
+    EXPECT_CALL(pm1, getPrice()).WillOnce(Return(mockFirstPrice));
+    EXPECT_CALL(pm1, getName()).WillOnce(Return(mockFirstName));
+    EXPECT_CALL(pm1, getBakingTime()).WillOnce(Return(mockFirstBakingTime));
+    EXPECT_CALL(pm2, getPrice()).WillOnce(Return(mockSecondPrice));
+    EXPECT_CALL(pm2, getName()).WillOnce(Return(mockSecondName));
+    EXPECT_CALL(pm2, getBakingTime()).WillOnce(Return(mockSecondBakingTime));
+    EXPECT_CALL(tm, sleep_for).Times(3);
```

Comment:
Not this way. You should expect that sleep_for is called with a proper duration - the baking time of each pizza.

Reply:
@ziobron could you explain? I think that is good solution